---
title: Joint Bayesian inference of bacterial ancestral recombination graphs
meeting: Phyloseminar
date: 2017-02-23
category: talk

layout: presentation

reveal:
    theme: theme.css
    slideNumber: true
    history: true
---

<section class="titlepage center">
    <h1 class="title">Joint Bayesian inference of bacterial ancestral recombination graphs</h1>

    <div class="authors">Tim&nbsp;Vaughan</div>

    <div class="institution">Centre for Computational Evolution<br>
        The University of Auckland</div>

    <div class="date">PhyloSeminar, 23<sup>rd</sup> February, 2017</div>

</section>

<section>
    <h2>Why study bacterial phylogenetics?</h2>

    <div style="float:left; width:700px">
        <ul>
            <li class="fragment" data-fragment-index="1">Bacteria play important roles (both positive and
                negative) in the health of animals and
                plants.</li>
            <li class="fragment" data-fragment-index="2">Many bacteria possess interesting and
                <i>experimentally accessible</i> evolutionary
                dynamics.</li>
            <li class="fragment" data-fragment-index="3">Bacterial genomes are measurably evolving over
                relatively short study periods.
                <div class="figure" style="width:500px; position: relative; left: 100px">
                    <img data-src="MEP_cropped.png">
                    <div class="cite">Drummond &amp; Rambaut, TIEE (2003)</div>
                </div>
            </li>
        </ul>
    </div>
    <div style="float:right; width:250px">
        <div class="figure fragment" data-fragment-index="1">
            <img data-src="EColiGut_Psa_collage.jpg">
        </div>
        <div class="figure fragment" data-fragment-index="2">
            <img data-src="adaptive_radiation.jpg">
            <div class="cite">Rainey &amp; Travisano, Nature (1998)</div>
        </div>
    </div>
</section>

<section>
    <h2>Cartoon bacterial population genetics</h2>

    <div style="position: relative; top:0; left:0; width:1000px; height:400px">
        <img data-src="binary_fission_7.svg" class="fragment current-visible" data-fragment-index="1"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_6.svg" class="fragment" data-fragment-index="2"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_5.svg" class="fragment" data-fragment-index="3"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_4.svg" class="fragment" data-fragment-index="4"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_3.svg" class="fragment" data-fragment-index="5"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_2.svg" class="fragment" data-fragment-index="6"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_1.svg" class="fragment" data-fragment-index="7"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
    </div>

    <img data-src="transduction_cropped.jpg" class="fragment" data-fragment-index="5"
                                                              style="position:absolute; top:100px; left:0px; height:250px;">

    <img data-src="conjugation.jpg" class="fragment" data-fragment-index="7"
                                                     style="position:absolute; top:100px; right:0px; height:250px;">
    <ul>
        <li class="fragment">Effect of events can be plasmid transfer, insertion or homologous recombination.</li>
        <li class="fragment">Rate of recombination varies and is subject to selection.</li>
        <li class="fragment">Focus on homologous recombination: only event which doesn't alter the length of the sequence</li>
    </ul>
</section>

<section>
    <h2>The Problem for Phylogenetic Inference</h2>

    <p>For many bacteria, the ratio between the recombination rate and the mutation rate is extremely high.</p>

    <div class="figure fragment">
        <img data-src="r2mhist.svg" style="width:800px">
        <div class="cite">Vos and Didelot, The ISME Journal (2009)</div>
    </div>
</section>

<section>
    <h2>Existing solutions</h2>

    <ul>
        <li class="fragment">Pre-processing of data to identify and remove
            non-vertically inherited material. (eg. START: <a class="cite" href="">Jolley et al., 2001</a>)

            <table style="width:800px">
                <tr><th style="width:400px">Pros</th> <th>Cons</th></tr>
                <tr>
                    <td>
                        <ul>
                            <li class="fragment">Can use standard tools for
                                phylogenetic inference.</li>
                        </ul>
                    </td>
                    <td>
                        <ul>
                            <li class="fragment">Data is being thrown away.</li>
                            <li class="fragment">Ad hoc, may bias results.</li>
                        </ul>
                    </td>
                </tr>

            </table>
        </li>

        <li class="fragment" style="margin-top:20px">Explicit modelling of bacterial recombination.<br>
            (eg. ClonalFrame and ClonalOrigin: <a class="cite" href="">Didelot et al., 2007</a>, <a class="cite" href="">2010</a>)

            <table style="width:800px">
                <tr><th style="width:400px">Pros</th> <th>Cons</th></tr>
                <tr>
                    <td>
                        <ul>
                            <li class="fragment">Can make use of all data.</li>
                            <li class="fragment">Can infer additional parameters such as recombination rates.</li>
                            <li class="fragment">May yield increased confidence in estimates.</li>
                        </ul>
                    </td>
                    <td>
                        <ul>
                            <li class="fragment">Models can be complex, with many parameters.</li>
                            <li class="fragment">Both computationally and statistically challenging.</li>
                            <li class="fragment">Existing implementations lack flexibility.</li>
                        </ul>
                    </td>
                </tr>
            </table>
        </li>
    </ul>

</section>

<section>
    <h2>The Coalescent with Gene Conversion</h2>

    <div style="position: relative; top:0; left:0; width:1000px; height:500px;">
        <img data-src="WrightFisherGC1.svg"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC2.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC3.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC3b.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC4b.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC5b.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
    </div>
    <a class="cite" href="">Wiuf, 1999</a>; <a class="cite" href="">Wiuf and Hein, 2000</a>

    <div class="fragment" style="position:absolute; top:100px; left:0px; width:350px">
        <table class="params">
            <tr><th colspan="2">Parameters</th></tr>
            <tr> <td>$N(t)g$</td> <td>Coalescence rate</td> </tr>
            <tr> <td>$\rho_s$</td> <td>Recombination rate</td> </tr>
            <tr> <td>$\delta$</td> <td>Expected tract length</td> </tr>
        </table>
    </div>
</section>

<section class="center">
    <div class="textbox alert">
        <div class="title">Problem</div>
        The space of possible ancestral recombination graphs is extremely
        large. Even two samples have infinitely many distinct ancestries!
    </div>
</section>

<section>
    <h2>Full ARG</h2>

    <img data-src="full_arg.svg" style="box-shadow:none;width:500px">

    <br>
    &nbsp;
</section>

<section>
    <h2>Approximation 1: ClonalFrame</h2>

    <img data-src="CF_arg.svg" style="box-shadow:none;width:500px">

    <br>

    <a class="cite" href="">Didelot et al., 2007 (ClonalFrame)</a>;
    <a class="cite" href="">Didelot and Wilson, 2015 (ClonalFrameML)</a>
</section>

<section>
    <h2>Approximation 2: ClonalOrigin</h2>

    <img data-src="CO_arg.svg" style="box-shadow:none;width:500px">

    <br>

    <a class="cite" href="">Didelot et al., 2010 (ClonalOrigin)</a>;
    <a class="cite" href="">Ansari and Didelot, 2014 (ClonalOrigin')</a>
</section>

<section class="center">
    <div class="textbox" style="width:600px;position:relative;left:200px">
        <div class="title">Our Goal</div>
        To conduct Bayesian inference of the full ARG under the ClonalOrigin
        model, optionally under non-trivial population size dynamics.
    </div>
</section>


<section>
    <h2>Inference under the ClonalOrigin Model</h2>

    <p> Inference follows the standard Bayesian phylogenetic tradition: </p>

    \begin{equation*}
        f(G,N,\mu,\rho,\delta|A) \propto P_{F}(A|G,\mu)f_{CGC}(G|N,\rho,\delta)f_{\text{prior}}(N
,\mu,\rho,\delta)
    \end{equation*}

    <p>where</p>

    <ul>
        <li>$A$ is the sequence alignment,</li>
        <li>$\mu$ are the substitution model parameters, and</li>
        <li>$G$ is the full sample genealogy including clonal frame<br>
            $T$ and $M$ conversions $\{C_i\}_{i\in[1\ldots M]}$.</li>
    </ul>

    <div class="fragment">
    <p>The genealogy density under ClonalOrigin model can be expanded</p>
    \begin{equation*}
        f_{CGC}(G|\rho',\delta,N)=\left(\prod_{i=1}^M f(C_i|T,N,\delta)\right)P(M|T,\rho)f_C(T|N)
    \end{equation*}
    </div>
</section>

<section>
    <h2>Identifiability</h2>

    <p>Despite using a simplified model, an infinite number of ARGs still
    possess the same likelihood given a sequence alignment.</p>

    <img class="fragment" data-src="identifiability.svg" style="box-shadow:none;width:800px">

    <blockquote class="fragment">
        Very important for an MCMC algorithm to propose state changes which minimize effect on likelihood.
    </blockquote>
</section>

<section>
    <h2>MCMC State Proposal Strategy</h2>

    <div style="position: relative; top:0; left:0; width:1000px; height:500px;">
        <img data-src="collapse_expand4.svg"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="collapse_expand3.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="collapse_expand2.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="collapse_expand1.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
    </div>

    <ul>
        <li class="fragment">Attachment points and site tracts of new conversions drawn from prior.</li>
    </ul>
</section>

<section>
    <h2>Validation 1: Sampling from the prior</h2>

    <ul>
        <li> It is trivial to sample directly from the approximate ARG prior
            via coalescent simulation.</li>
        <li class="fragment"> We expect precise agreement between our MCMC algorithm and the
            coalescent simulation when no data is involved.</li>
    </ul>

    <div class="fragment">
        <img data-src="validate_CFheight.svg" style="box-shadow:none">
        <img data-src="validate_meanEdgeLength.svg" style="box-shadow:none">
        <img data-src="validate_meanTractLength.svg" style="box-shadow:none">
    </div>

    <blockquote class="alert fragment">
        One should alwayas test that a new sampler exactly converges to a known
        distribution: this provides a strict necessary criterion for
        correctness.
    </blockquote>
</section>

<section>
    <h2>Validation 2: Inference from simulated data</h2>

    <img data-src="posterior_rho.svg" style="box-shadow:none; padding:0; margin:0">
    <img data-src="posterior_delta.svg" style="box-shadow:none;padding:0;margin:0">
    <img data-src="posterior_theta.svg" style="box-shadow:none;padding:0;margin:0">
</section>

<section>
    <p style="margin:0px">True ARG:</p>
    <img data-src="true_tree.svg" style="box-shadow:none; padding:0; margin:0; width:800px">

    <div class="fragment">
        <p style="margin:0px">Randomly-selected ARG from MCMC:</p>
        <img data-src="sampled_tree.svg" style="box-shadow:none; padding:0; margin:0; width:800px">
    </div>
</section>

<section>
    <h2>Summary networks</h2>

    <div style="position: relative; top:0; left:0; width:1000px; height:500px;">
        <img data-src="summarizing6.svg"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing5.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing3.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing2.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing1.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing0.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
    </div>
</section>

<section>
    <p style="margin:0px">True ARG:</p>
    <img data-src="true_tree.svg" style="box-shadow:none; padding:0; margin:0; width:800px">

    <div class="fragment">
        <p style="margin:0px">Summary ARG from MCMC:</p>
        <img data-src="summary_tree.svg" style="box-shadow:none; padding:0; margin:0; width:800px">
    </div>
</section>

<section>
    <h2>Application to <i>Escherichia coli</i></h2>

    <ul>
        <li>Applied method to <i>E. coli</i> O157/O26 sequences collected by Tessy George (<sup>m</sup>EpiLab, Massey University, New Zealand).</li>
        <li>Sequences for 53 rMLST genes (21.1 kb total) from each of 23 human and cattle-derived isolates.</li>
        <li>Analyzed using fixed expected tract length parameter and informative prior on conversion rate parameter.</li>
    </ul>

    <img data-src="Tessy_rMLST_splitstree.svg" style="width:900px;box-shadow:none">
</section>

<section>
    <h2>Application to <i>Escherichia coli</i></h2>

    <div style="position: relative; top:0; left:0; width:1000px; height:600px;">
        <img data-src="Tessy_new_summary2.svg" style="position:absolute;top:0;left:0;width:1000px;box-shadow:none;background-color:white">
        <img class="fragment" data-src="Tessy_new_summary2_compare.svg" style="position:absolute;top:0;left:0;width:1000px;box-shadow:none;background-color:white">
        <img class="fragment" data-src="gene_flow.svg" style="position:absolute;top:0;left:0;width:250px;background-color:white;padding:20px">
    </div>
</section>

<section>
    <h2>Software implementation</h2>

    <img data-src="bacter_logo.svg" style="width:500px;box-shadow:none">

    <br>

    <a href="http://tgvaughan.github.io/bacter">tgvaughan.github.io/bacter</a>

    <br>
    <br>

    <ul>
        <li class="fragment">BEAST 2 package that performs inference under the ClonalOrigin model.</li>
        <li class="fragment">Joint inference of the clonal frame, recombinant edges and parameters.</li>
        <li class="fragment">Can be combined with usual variety of substitution models, parametric population models and parameter priors.</li>
        <li class="fragment">GUI analysis configuration via BEAUti.</li>
    </ul>
</section>

<section>
    <h2>Recombination and Demographic Inference</h2>

    <br>
    <br>

    <img data-src="conversion_alignment.png" style="width:800px">

    <ul>
        <li class="fragment">Relationship used by
            <a class="cite" href="http://dx.doi.org/10.1038/nature10231">Li and Durbin, 2011</a>
            to infer human demographic history from pairs of autosomes.</li>
    </ul>
</section>

<section>
    <h2>Bayesian Skyline Plots in Bacter</h2>

    <img data-src="skyline_icon.svg" style="box-shadow:none;width:400px">

    <ul>
        <li class="fragment">Bacter implements a non-parametric demographic history model inspired by the standard Bayesian Skyline Plot <a href="https://doi.org/10.1093/molbev/msi103">(Drummond et al., 2005)</a>.</li>
        <li class="fragment">In principle can reconstruct non-trivial dynamics from very few genomes.</li>
        <li class="fragment">In practice, this is limited by the computational expense of the algorithm.</li>
    </ul>
</section>

<section>
    <h2>Piecewise constant/linear population model</h2>

    <div style="position: relative; top:0; left:0; width:1000px; height:600px;">
        <img data-src="skyline_diagram.svg" style="position:absolute;left:0;top:0;width:1000px;box-shadow:none">
        <img class="fragment" data-src="skyline_linear_diagram.svg" style="position:absolute;left:0;top:0;width:1000px;box-shadow:none;background-color:white">
    </div>
</section>

<section>
    <h2>Inference example</h2>

    <img data-src="skyline_tree.svg" style="width:700px;box-shadow:none;margin:0">


    <img class="fragment" data-src="popfun_inf.svg" style="width:800px;margin-top:0">
</section>

<section>
     <h2>Bacter limitations</h2>

     <div style="float:left;width:600px">
     <ul>
         <li class="fragment">Computational complexity scales with the number of proposed conversions.
             <ul>
                 <li class="fragment">This can be arbitrarily large, even for small sample sets.</li>
                 <li class="fragment">Cannot use the BEAGLE phylogenetic likelihood library to speed things up due
                     to peculiarities of ARG likelihood computation.</li>
             </ul>
             <br>
         </li>
         <li class="fragment">MCMC algorithm used does not intelligently locate conversions.
             <ul>
                 <li class="fragment">Looking at addressing this in the near future.</li>
             </ul>
             <br>
         </li>

         <li class="fragment">Network summary algorithm can produce peculiar results.
             <ul>
                 <li>More research needs to be done to find a better algorithm.</li>
             </ul>
         </li>
     </ul>
     </div>
     <div style="float:right">
         <br>
         <br>
         <img data-src="ointment.svg" style="width:300px;box-shadow:none;"/>
     </div>
</section>

<section>
    <h2>Acknowledgements</h2>

    <ul>
        <li>Members of <sup><i>m</i></sup>EpiLab, Massey University, New Zealand
            <ul>
                <li>Patrick Biggs, Tessy George and Nigel French (providing hard-won data)</li>
            </ul>
            <br>
        </li>
        <li>Members of the Centre for Computational Evolution, University of Auckland, New Zealand
            <br>
            <br>
        </li>
        <li>The following organizations:<br><br>
            <div style="text-align:center">
                <img data-src="UoA_logo.svg" style="box-shadow:none;height:80px">
                <img data-src="Massey_logo.svg" style="box-shadow:none;height:80px;padding-left:50px"><br>
                <img data-src="AWC_logo.svg" style="box-shadow:none;height:80px">
                <img data-src="MF_logo.png" style="box-shadow:none;height:80px;padding-left:100px">
            </div>
        </li>
    </ul>

    <blockquote>
        Slides of this talk are online at<br>
        <a style="color:white" href="https://tgvaughan.github.io/talks/PhyloSeminar17">tgvaughan.github.io/talks/PhyloSeminar17</a>
    </blockquote>
</section>
