<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Computational Phylogenetics using BEAST 2</title>

		<meta name="description" content="Computational Phylogenetics using BEAST 2">
		<meta name="author" content="Tim Vaughan">

		<meta name="apple-mobile-web-app-capable" content="yes">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">

		<link rel="stylesheet" href="../reveal.js/css/reveal.css">
		<link rel="stylesheet" href="wb_theme.css" id="theme">

		<!-- Code syntax highlighting -->
		<link rel="stylesheet" href="../reveal.js/lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../reveal.js/css/print/pdf.css' : '../reveal.js/css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
			<div class="slides">
				<section class="titlepage">
                    <div class="title">
                        <h3>Computational Phylogenetics using BEAST 2</h3>
                    </div>

                    <div class="authors">Tim&nbsp;Vaughan and David&nbsp;Welch</div>
                    <div class="institution">
                        <img data-src="UoA_logo.svg" width="250"/><br>
                        Centre for Computational Evolution
                    </div>

                    <div class="meeting">Winter Bootcamp 2016</div>
				</section>

                <section>
                    <section class="center">
                        <h2>Biology</h2>
                    </section>

                    <section>
                        <h3>What is evolution?</h3>

                        <br>

                        <blockquote class="fragment"> Evolution is change in heritable traits of
                            biological populations over successive generations.
                        </blockquote>

                        <br>

                        <blockquote class="fragment">Evolutionary processes give rise to biodiversity at
                            every level of biological organisation, including
                            <b>species</b>, individual <b>organisms</b>,
                            and <b>genes</b>.</blockquote>

                    </section>

                    <section>
                        <h3>What is a phylogenetic tree?</h3>
                        <div style="position:relative; width:800px; height:600px; margin:0 auto;">
                            <div class="fragment current-visible"
                                style="position:absolute">

                                <br>

                                <blockquote> A phylogenetic tree is a
                                    branching diagram or "tree" showing the evoluionary
                                    relationships between various species, individuals,
                                    or genes.
                                </blockquote>
                            </div>
                            <img class="fragment current-visible" width="600" style="margin:0px;padding:0px;"
                                data-src="primate_tree.jpg"
                                style="position:absolute;top:0;left:0;" />
                            <div class="figure fragment"
                                style="position:absolute;top:0;left:0;margin:0;padding:0;">
                                <img width="800" data-src="flu_tree.png" style="margin:0px;padding:0px"/>
                                <div class="cite">Bedford et al., PLoS Pathogens, 2010</div>
                            </div>
                        </div>
                    </section>

                    <section>
                        <h3>Evolutionary processes</h3>

                        Evolution is governed by three main processes:

                        <ol>
                            <li class="fragment"> <b>Reproduction</b> </li>
                            <ul class="fragment">
                                <li> Process by which heritable traits are passed on to the next generation. </li>
                            </ul>
                            <li class="fragment"> <b>Mutation</b> </li>
                            <ul>
                                <li class="fragment"> Usually occurs due to noise in reproductive process. </li>
                                <li class="fragment"> Provides variability in traits. </li>
                            </ul>
                            <li class="fragment"> <b>Selection</b> </li>
                            <ul class="fragment">
                                <li> Probability of individual reproducing may depend (weakly or strongly) on value of heritable trait. </li>
                            </ul>
                        </ol>

                        <p class="fragment"><i>Populations evolve, individuals do not!</i></p>
                    </section>

                    <section>
                        <h3>Genetic drift</h3>

                        <blockquote>The random fluctuation in the frequency of
                        a trait or allele (gene variant) distribution within a
                        population due to the demographic process.</blockquote>


                        <img class="fragment" style="vertical-align: middle; width: 400px; margin-right: 20px" data-src="drift1.svg"/>
                        <img class="fragment" style="vertical-align: middle; width: 400px" data-src="drift2.svg"/>
                    </section>

                    <section>
                        <h3>Modelling neutral genetic drift</h3>

                        <!-- Wright-Fisher model -->
                        <div style="position:relative; width:800px; height:400px; margin:0 auto;">
                            <img class="fragment current-visible" width="800" data-src="WF5.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                            <img class="fragment current-visible" width="800" data-src="WF4.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                            <img class="fragment current-visible" width="800" data-src="WF3.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                            <img class="fragment current-visible" width="800" data-src="WF2.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                            <img class="fragment" width="800" data-src="WF.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                        </div>
                        <ul>
                            <li class="fragment">All individuals descend from a common ancestor at some point in the past.</li>
                            <li class="fragment">Probability that the descendants of a given individual "take over" is $1/2N$.</li>
                        </ul>

                    </section>

                    <section>
                        <h3>The molecular clock hypothesis</h3>

                        <br>

                        <ol>
                            <li class="fragment">In a single generation a new allele is produced in the population with probability $2N\mu$.</li>
                            <li class="fragment">Any individual gene eventually becomes fixed in the population with probability $1/2N$.
                        </ol>

                        <br>
                        <br>

                        <blockquote class="fragment">
                        The probability per generation that a new allele fixes
                        is therefore $\mu$: independent of population size!
                        </blockquote>

                        <br>

                        <blockquote class="alert fragment">This is not exactly true.</blockquote>

                    </section>

                    <section>
                        <h3>The phylogenetic inference problem</h3>

                        <blockquote>Given a set of $M$ sampled molecular
                        sequences, infer the evolutionary
                        relationships between them.</blockquote>

                        <img class="fragment" data-src="primate_align.png"/>

                        <p class="fragment">In a sense this is a just a clustering problem...</p>

                    </section>

                    <section>
                        <h3>The phylogenetic inference problem</h3>

                        <img data-src="primate_nj.svg" height="450"/>

                        <ul>
                            <li class="fragment">Edge lengths may represent genetic distance  or time (if the molecular clock rate is known).</li>
                            <li class="fragment">How can we do better than this?</li>
                        </ul>
                    </section>

                    <section>
                        <h3>The genealogical process</h3>

                        <ul>
                            <li>In the absence of selection, we can consider the process of tree generation in isolation.</li>
                            <li class="fragment">There are numerous models that apply for different phylogenetic trees, including:
                                <ol>
                                    <li class="fragment">The <b>Yule process</b> as a simple model for species tree generation, and</li>
                                    <li class="fragment">the <b>coalescent process</b> as a simple model for gene tree generation.</li>
                                </ol>
                            </li>
                        </ul>
                    </section>

                    <section>
                        <h3>The Yule process</h3>
                        Assumes a constant rate of speciation through time:

                        <img class="fragment" data-src="yule_tree.svg" height="450"> <br>

                        <p class="fragment">Single parameter is the speciation rate $\lambda$.</p>
                    </section>

                    <section>
                        <h3>The coalescent process</h3>
                        Arises as a limit of many models in population genetics.

                        <div style="position:relative; width:800px; height:450px; margin:0 auto;">
                            <img class="fragment current-visible" width="800" data-src="Coalescent.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                            <img class="fragment current-visible" width="800" data-src="Coalescent2.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                            <img class="fragment current-visible" width="800" data-src="Coalescent3.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                            <img class="fragment" width="800" data-src="Coalescent4.svg"
                                 style="margin:0px;padding:0px;position:absolute;top:0;left:0;box-shadow:none;"/>
                        </div>

                        <ul>
                            <li class="fragment">Probability of coalescence in single generation  $\sim\binom{k}{2}\frac{1}{N}$</li>
                            <li class="fragment">Geometric distribution between coalescence events.</li>
                        </ul>
                    </section>
                    <section>
                        <h3>Example coalescent tree</h3>

                        <br>

                        <img data-src="coalescent_tree.svg" height="450"> <br>
                    </section>

                    <section>
                        <h3>Coalescent trees and population dynamics</h3>

                        The coalescent rate is inversely proportional to the (effective) population size.

                        <div class="figure">
                            <img data-src="mep.png" height="450">
                            <div class="cite">Drummond et al., Trends in Ecology and Evolution, 2003</div>
                        </div>
                    </section>

                    <section class="center">
                        <blockquote>How do we go about combining all of these different pieces of information?</blockquote>

                        <br><br>
                        <p class="fragment">The answer is...</p>
                    </section>
                </section>

                <section>
                    <section class="center">
                        <h2>Bayesian statistics</h2>
                    </section>

                    <section>
                        <h3>The meaning of Probability</h3>

                        We use the following definition of probability:
                        <blockquote class="fragment">For propositions $A$ and $B$, the probability
                            $P(A|B)$ is the degree to which $A$ is
                            believed to be true,<br>on the condition that $B$ is true.</blockquote>

                        <ul>
                            <li class="fragment">Who is doing the believing?</li>
                            <li class="fragment">Probability is subjective.</li>
                        </ul>

                        <img class="fragment" data-src="dice_inference.svg" width="800" style="box-shadow:none">
                    </section>

                    <section>
                        <h3>Manipulating probabilities</h3>

                        There are only two rules for manipulating probabilities:<br>

                        <br>

                        <ol>
                            <li class="fragment">The product rule:
                                $$P(A|B,C)P(B|C)=P(A,B|C)$$
                                where $A,B$ represents $A$ <i>and</i> $B$.
                            </li>
                            <br>

                            <li class="fragment">The sum rule:
                                $$P(A|C) + P(B|C) - P(A,B|C) = P(A\vee B|C)$$
                                where $A\vee B$ represents $A$ <i>or</i> $B$.
                            </li>
                        </ol>

                        <br><br>
                        <p class="fragment">That's it! You can now do Bayesian statistics.</p>
                    </section>

                    <section>
                        <h3>Bayes' rule</h3>

                        Suppose we have a probabilistic model $M$ with
                        parameters $\theta_M$.  Given data $D$ which we
                        assume has been generated by the model, what can we learn about the parameters?

                        <br><br>

                        <ul>
                            <li class="fragment">Our model allows us to evaluate $P(D|M,\theta_M)$.</li>
                            <li class="fragment">We want to know $P(\theta_M|D,M)$.</li>
                        </ul>

                        <br><br>

                        <p class="fragment">Mechanically applying the product rule yields</p>
                        <blockquote class="fragment">
                            $$\color{darkred}{P(\theta_M|D,M)}=\frac{\color{yellow}{P(D|M,\theta_M)}\color{darkblue}{P(\theta_M|M)}}{P(D|M)}$$
                        </blockquote>
                        <p class="fragment">with terms named the <b style="color:darkred">posterior</b>, <b style="color:#808000">likelihood</b> and <b style="color:darkblue">prior</b>.</p>
                    </section>

                    <section>
                        <h3>Bayesian inference</h3>

                        Bayes' rule gives us a natural framework for drawing on many sources of information:

                        <br><br>

                        <ul>
                            <li class="fragment">$\color{darkblue}{P(\theta|M_{\theta})}$
                                describes our state of knowledge
                                <b style="color:darkblue">prior</b> to
                                receiving $D$. (May still depend on expert
                                knowledge.)</li>
                            <li class="fragment">$\color{#808000}{P(D|\theta,M_{\theta})}$ is
                                the <b style="color:#808000">likelihood</b> of
                                $\theta$ given $D$. Describes how the data
                                modifies our knowledge of $\theta$.</li>
                            <li class="fragment">$\color{darkred}{P(\theta|D, M_{\theta})}$
                                describes our state of knowledge
                                <b style="color:darkred">after</b> receiving $D$.</li>
                        </ul>

                        <br><br>
                        <i class="fragment">The posterior of one analysis may be the prior of a second.</i>
                    </section>

                </section>

                <section>
                    <section class="center">
                        <h2>Bayesian phylogenetic inference</h2>
                    </section>

                    <section>
                        <h3>Goals of Bayesian phylogenetic inference</h3>

                        <p>The goal of phylogenetic inference is generally to
                        learn about the phylogeny of a set of sampled genetic
                        sequences.</p>

                        <p class="fragment">As Bayesians we accomplish this by seeking to characterise the joint posterior:
                        $$P(T,\mu,\theta|D) = \frac{1}{P(D)} P(D|T,\mu)P(T|\theta)P(\mu)P(\theta)$$</p>

                        <ul class="fragment">
                            <li>$P(D|T,\mu)$ is the <i>tree likelihood</i></li>
                            <li>$P(T|\theta)$ is the <i>tree prior</i></li>
                            <li>$P(\mu)$ and $P(\theta)$ are the <i>parameter priors</i></li>
                        </ul>

                        <br><br>

                        <i class="fragment">We simultaneously learn about the tree and the parameters!</i>

                    </section>

                    <!-- Likelihood -->
                    <section>
                        <h3>The Tree Likelihood</h3>

                        <blockquote>The tree likelihood $P(D|T,\mu)$ is the probability that the observed sequences were produced by evolving a single ancestral sequence down $T$ under a neutral substitution model with parameters $\mu$.</blockquote>

                        <img class="fragment" data-src="likelihood.svg" width="800">

                        <p class="fragment">There is an efficient algorithm for computing this
                        (summing over all possible ancestral sequences):
                        <b>Felsenstein's pruning algorithm</b>.</p>
                    </section>

                    <!-- Tree priors -->
                    <section>
                        <h3>The Tree Prior</h3>

                        <blockquote>The tree prior $P(T|\theta)$ is the
                            probability for the tree $T$ based solely on the
                            genealogical model and its parameters $\theta$.<br>
                        </blockquote>

                        <p class="fragment">(May also be considered the likelihood for the
                        genealogical model parameters $\theta$ given the tree.)</p>

                        <ul>
                            <li class="fragment">Yule prior (speciation rate $\lambda$)</li>
                            <li class="fragment">Coalescent prior (population function $N(t)$)</li>
                            <li class="fragment">Structured Coalescent prior (sub-population sizes $N_i$, migration rates $m_{ij}$))</li>
                            <li class="fragment">Birth-death priors (birth and death rates)</li>
                            <li class="fragment">Transmission tree priors (infection, recovery rates)</li>
                            <li class="fragment">...</li>
                        </ul>

                    </section>

                    <!-- Serial vs contemporaneously sampled sequences -->
                    <!--section>
                        <h3>Sampling sequences through time</h3>
                    </section-->

                    <!-- number of trees -->
                    <section>
                        <h3>How many trees are there?</h3>

                        <div class="figure inline" style="width:550px">
                            <img data-src="howmanytrees.png" style="height:550px"> 
                            <div class="cite">&nbsp;</div>
                        </div>

                        <div class="figure inline" style="width:400px">
                            <img data-src="numtrees.png" style="height:550px"> 
                            <div class="cite">Felsenstein, 2004</div>
                        </div>
                    </section>
                </section>

                <section>
                    <section class="center">
                        <h2>Markov Chain Monte Carlo (MCMC)</h2>
                    </section>

                    <section>
                        <h3>MCMC basics</h3>

                        <p>MCMC (aka the Metropolis-Hastings algorithm) allows us
                        to generate <b>samples</b> from a target distribution
                        $P(\vec{x})$ assuming that we can evaluate $\alpha
                        P(\vec{x})$ where $\alpha$ is an arbitrary unknown
                        constant.</p>

                        <br>

                        <p class="fragment">MCMC works by performing a biased random walk over
                        $\vec{x}$ constructed so that the relative frequency of
                        visiting state $\vec{x}$ at equilibrium is equal to
                        $P(\vec{x})$.</p>

                        <br>

                        <blockquote class="alert fragment">Flexible, yet extremely computationally intensive!</blockquote>
                    </section>

                    <section>
                        <h3>MCMC robot</h3>

                        <div class="figure">
                            <img data-src="MCMCrobot1.png" width="800">
                            <div class="cite">Paul Lewis</div>
                        </div>
                    </section>

                    <section>
                        <h3>MCMC robot</h3>

                        <div class="figure">
                            <img data-src="MCMCrobot2.png" width="800">
                            <div class="cite">Paul Lewis</div>
                        </div>
                    </section>

                    <section>
                        <h3>Pure random walk</h3>

                        <div class="figure">
                            <img data-src="MCMCrobot3.png" width="800">
                            <div class="cite">Paul Lewis</div>
                        </div>
                    </section>

                    <section>
                        <h3>Burn-in</h3>

                        <div class="figure">
                            <img data-src="MCMCrobot4.png" width="800">
                            <div class="cite">Paul Lewis</div>
                        </div>
                    </section>

                    <section>
                        <h3>Convergence</h3>

                        <div class="figure">
                            <img data-src="MCMCrobot5.png" width="800">
                            <div class="cite">Paul Lewis</div>
                        </div>
                    </section>

                </section>

                <section>
                    <section class="center">
                        <h2>BEAST</h2>
                    </section>

                    <section>
                        <img data-src="beast2_logo.png" style="box-shadow:none">

                        <blockquote>BEAST is a free (as in freedom) software
                            package that uses MCMC to perform Bayesian
                            phylogenetic inference.</blockquote>

                        <br>
                        <p>Project website: <a href="http://www.beast2.org">beast2.org</a></p>
                        <ul>
                            <li>Download software</li>
                            <li>Documentation and FAQs</li>
                            <li>Tutorials</li>
                            <li>...</li>
                        </ul>

                    </section>

                    <section>
                        <h3>BEAST components</h3>

                        <ul>
                            <li><b>BEAST2</b>: Software implementing MCMC for model parameter and tree inference</li>
                            <li><b>BEAUti</b>: GUI for setting up the input file</li>
                            <li><b>Tracer</b>: Tool for summarizing parameter posteriors</li>
                            <li><b>Tree Annotator</b>: Tool for summarizing tree posteriors</li>
                            <li><b>Figtree</b>: Tool for visualising trees</li>
                        </ul>
                    </section>

                    <section>
                        <h3>BEAST Workflow</h3>

                        <img data-src="workflow.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="beauti1.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="beauti2.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="beauti3.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="beauti4.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="beast2slide1.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="beast2slide2.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="tracerslide1.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="tracerslide2.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="tracerslide3.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="tracerslide4.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="TAslide1.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="TAslide2.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="FTslide1.png" style="box-shadow:none">
                    </section>

                    <section>
                        <img data-src="FTslide2.png" style="box-shadow:none">
                    </section>
                </section>

                <section>
                    <section class="center">
                        <h2>BEAST Tutorials</h2>
                    </section>

                    <section>
                        <h3>Tutorials</h3>

                        <br><br>

                        <p>Tutorial 1:</p>
                        <blockquote>Divergence time dating.</blockquote>

                        <br>

                        <p>Tutorial 2:</p>
                        <blockquote>Bayesian skyline plot.</blockquote>
                        <!-- Bayesian skyline plot -->
                    </section>

                    <section>
                        <h3>Accessing the tutorials</h3>
                        <div class="figure">
                            <img data-src="beast2tutorials.png">
                            <div class="cite">www.beast2.org/tutorials</div>
                        </div>
                    </section>
                </section>
			</div>

		</div>

		<script src="../reveal.js/lib/js/head.min.js"></script>
		<script src="../reveal.js/js/reveal.js"></script>

		<script>

			// Full list of configuration options available at:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: false,
				history: true,
				center: false,
                mouseWheel: true,

				transition: 'convex', // none/fade/slide/convex/concave/zoom

                math: {
                    mathjax: 'https://cdn.mathjax.org/mathjax/latest/MathJax.js',
                    config: 'TeX-AMS_HTML-full'  // See http://docs.mathjax.org/en/latest/config-files.html
                },

				// Optional reveal.js plugins
				dependencies: [
					{ src: '../reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: '../reveal.js/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: '../reveal.js/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: '../reveal.js/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: '../reveal.js/plugin/zoom-js/zoom.js', async: true },
					{ src: '../reveal.js/plugin/notes/notes.js', async: true },
                    { src: '../reveal.js/plugin/math/math.js', async: true }
				]
			});

		</script>

	</body>
</html>
